executor/api/rest/server.go

```diff
@@ -141,9 +141,15 @@ func (r *SeldonRestApi) Initialise() {
 
 func puidHeader(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if puid := r.Header.Get(payload.SeldonPUIDHeader); puid == "" {
-			r.Header.Set(payload.SeldonPUIDHeader, guuid.New().String())
+		puid := r.Header.Get(payload.SeldonPUIDHeader)
+		if len(puid) == 0 {
+			puid = guuid.New().String()
+			r.Header.Set(payload.SeldonPUIDHeader, puid)
 		}
+		if req_puid := w.Header().Get(payload.SeldonPUIDHeader); len(req_puid) == 0 {
+			w.Header().Set(payload.SeldonPUIDHeader, puid)
```

Comment:
Is there any case where we want the `SeldonPUIDHeader` of the response to differ from the request? I would just write it on the response and override it if there was any other. 

What are your thoughts @cliveseldon @axsaucedo?

Reply:
Oh I see - actually now that I think about it, I think this may be something that we want to always set as opposed to allowing users to set it when sending the request. The reason why , is that it's always better to ensure that the uuid is always unique for the request that is sent, and if they wanted to send an identified, they can send it on the metadata.

Reply:
This would become even more important if we're looking to do async in the future, we would need to make sure our UUIDs are always unique, and hence have to be generated always from inside - if external callers want to be able to find the response of the initial request they would be able to send an identifier as part of the metadata so they are able to find it again

Reply:
@cliveseldon what's your thoughts on this?

Reply:
I think its better to allow users to have control as they may wish to have their own UID scheme. 
If its clear in the docs. Most users won't set it.

Reply:
Hmm ok, I think as we're currently still synchronous it makes sense, but it does become quite key if we have asynchronous to no allow to set it as we can't enforce external UUIDs are unique, ideally we'd always create one to make sure we know that specific request sent was unique, and clients can also send their custom one which allows them to match them.

Reply:
In that case then it seems the PR should be done - would there be any further changes required?

Reply:
nit: should it not be `res_puid` for the var name?

Reply:
haha fair enough